### PR TITLE
Check for packages.config in pure C# updater.

### PR DESCRIPTION
**Pull request by sebasgomez238**
Last week we added functionality in the ruby code to check for a packages.config file in the same directory of each csproj. I confirmed this change correctly updates all relevant packages.config files relevant to the update #10803.

This PR adds the same functionality to the C# updater to ensure we don't regress when it is ready to be used.

### CODE REVIEW

File: nuget/helpers/lib/NuGetUpdater/NuGetUpdater.Core/Run/RunWorker.cs [hunk 1]
```diff
@@ -122,7 +122,7 @@ private async Task<RunResult> RunForDirectory(Job job, DirectoryInfo repoContent
         _logger.Log(JsonSerializer.Serialize(discoveryResult, DiscoveryWorker.SerializerOptions));
 
         // report dependencies
-        var discoveredUpdatedDependencies = GetUpdatedDependencyListFromDiscovery(discoveryResult);
+        var discoveredUpdatedDependencies = GetUpdatedDependencyListFromDiscovery(discoveryResult, repoDirectory);
```
Comment (sebasgomez238) [inline nuget/helpers/lib/NuGetUpdater/NuGetUpdater.Core/Run/RunWorker.cs:125]: Still need to verify this the correct variable to pass when running e2e.

Comment (sebasgomez238) [reply nuget/helpers/lib/NuGetUpdater/NuGetUpdater.Core/Run/RunWorker.cs]: Resolving. Based on RunWorker unit tests repoContentsPath.FullName gives me the path I want here.

**Review by brettfo — CHANGES_REQUESTED** [review]

File: nuget/helpers/lib/NuGetUpdater/NuGetUpdater.Core.Test/Run/UpdatedDependencyListTests.cs [hunk 2]
```diff
@@ -32,7 +65,7 @@ public void GetUpdatedDependencyListFromDiscovery()
                 }
             ]
         };
-        var updatedDependencyList = RunWorker.GetUpdatedDependencyListFromDiscovery(discovery);
+        var updatedDependencyList = RunWorker.GetUpdatedDependencyListFromDiscovery(discovery, path_to_contents: temp.DirectoryPath);
```
[the remark this answers is not in the document]
Comment (sebasgomez238) [reply nuget/helpers/lib/NuGetUpdater/NuGetUpdater.Core.Test/Run/UpdatedDependencyListTests.cs:68]: Ah sorry, using VS Code makes me feel like I'm using python again.

File: nuget/helpers/lib/NuGetUpdater/NuGetUpdater.Core/Run/RunWorker.cs [hunk 3]
```diff
@@ -294,6 +315,17 @@ string GetFullRepoPath(string path)
             auxiliaryFiles.Add(GetFullRepoPath(discoveryResult.DirectoryPackagesProps.FilePath));
         }
 
+        foreach (var project in discoveryResult.Projects)
+        {
+            var projectDirectory = Path.GetDirectoryName(project.FilePath);
+            var pathToPackagesConfig = Path.Join(pathToContents, "src", projectDirectory, "packages.config").NormalizePathToUnix().EnsurePrefix("/");
```
Comment (sebasgomez238) [inline nuget/helpers/lib/NuGetUpdater/NuGetUpdater.Core/Run/RunWorker.cs:321]: Reminder to remove this hard coded "src"

File: nuget/helpers/lib/NuGetUpdater/NuGetUpdater.Core/Run/RunWorker.cs [hunk 4]
```diff
@@ -182,7 +192,7 @@ await _apiHandler.IncrementMetric(new()
                     {
                         // TODO: this is inefficient, but not likely causing a bottleneck
                         var previousDependency = discoveredUpdatedDependencies.Dependencies
-                            .Single(d => d.Name == dependency.Name && d.Requirements.Single().File == Path.Join(discoveryResult.Path, project.FilePath).NormalizePathToUnix().EnsurePrefix("/"));
+                            .Single(d => d.Name == dependency.Name);
```
Comment (sebasgomez238) [inline nuget/helpers/lib/NuGetUpdater/NuGetUpdater.Core/Run/RunWorker.cs:195]: Still need to investigate more on whether removing this check is okay or if there is a way to check for the correct file. Will be done before completing this PR.

File: nuget/helpers/lib/NuGetUpdater/NuGetUpdater.Core/Run/RunWorker.cs [hunk 5]
```diff
@@ -305,6 +359,16 @@ string GetFullRepoPath(string path)
         return updatedDependencyList;
     }
 
+    private static string GetPackagesConfigFromCsprojLocation(string fullRepoPath)
+    {
+        int lastSlash;
+        fullRepoPath = Path.GetFullPath(fullRepoPath).NormalizePathToUnix().EnsurePrefix("/");
+
+        lastSlash = fullRepoPath.LastIndexOf('/');
```
Comment (brettfo) [inline nuget/helpers/lib/NuGetUpdater/NuGetUpdater.Core/Run/RunWorker.cs:367]: You can use `Path.GetDirectoryName()` to do this work for you.  Something like `Path.Combine(Path.GetDirectoryName(fullRepoPath), "packages.config")`

**Review by brettfo — APPROVED** [review]

**Pull request merged**